devices/r4isdhc.cpp

```diff
@@ -171,41 +171,52 @@ bool writeNor(const uint32_t dest_address, const uint32_t length, const uint8_t
     return true;
 }
 
-bool sendAndVerifyRawInit() {
-    if (ntrcard::state.status != ntrcard::Status::RAW) {
-        logMessage(LOG_ERR, "r4isdhc: sendAndVerifyRawInit: status (%d) not RAW",
-            static_cast<uint32_t>(ntrcard::state.status));
-        return false;
-    }
+bool checkCartType1() {
+    ntrcard::Status orig_status = ntrcard::state.status;
 
     CmdBuf4 buf;
     // this is actually the NOR write disable command
     // the r4isdhc will respond to cart commands with 0xFFFFFFFF if
     // the "magic" command hasn't been sent, so we check for that
     sendCommand(0x40199, 4, buf.u8, 0x180000);
-    if (buf.u32 != 0xFFFFFFFF) {
-        logMessage(LOG_ERR, "r4isdhc: sendAndVerifyRawInit: pre-test returned 0x%08X", buf.u32);
-        return false;
+    if (ntrcard::state.status == ntrcard::Status::RAW) {
+        if (buf.u32 != 0xFFFFFFFF) {
+            logMessage(LOG_ERR, "r4isdhc: checkCartType1: pre-test returned 0x%08X", buf.u32);
+            return false;
+        }
+    } else {
+        if (buf.u32 == 0) {
```

Comment:
This check here is useless, because what is probably happening is that the R4isdhc is returning either 0 or 0xFFFFFFFF in RAW (not sure which), but the DS applies the KEY2 xorstream on it, and we end up with garbage.

Reply:
(I'd rather just not have this check &mdash; the real check is on L208 anyway)